# Основы NumPy и Pandas
>_Изучите теоретическую часть!_

Модернизируйте заготовку заменив все участки ```<put your code here>``` на ваш код для того, что бы:
* пропускать первую строку c заголовком для файла sample_data/california_housing_test.csv
* приводить нужные колонки (по индексу) к типу с плавающей запятой (float)
* вычислять средние значени в колонках: housing_median_age, total_rooms, population

In [2]:
import numpy as np
import pandas as pd
import html5lib
titanic = pd.read_html('https://github.com/ogrisel/parallel_ml_tutorial/blob/master/notebooks/titanic_train.csv', header = 0, index_col='PassengerId')[0]
titanic = titanic.drop(titanic.columns[0], axis=1) # уберем не информативную колонку
titanic.head()

Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
1                1      0         A/5 21171   7.2500   NaN        S  
2                1      0          PC 17599  71.2833   C85        C  
3                0      0  STON/O2. 3101282   7.9250   NaN        S  
4                1      0            113803  53.1000  C123        S  
5                0      0            373450   8.0500   NaN        S

In [3]:
titanic.describe()

Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

**Для примера отберем пассажиров, которые сели в Cherbourg (Embarked=C) и заплатили более 200 у.е. за билет (fare > 200).**

Убедитесь, что Вы понимаете, как эта конструкция работает.
Если нет – посмотрите, как вычисляется выражение в квадратных в скобках.

In [ ]:
titanic[(titanic["Embarked"] == "C") & (titanic.Fare > 200)].head()

Survived  Pclass  ...            Cabin Embarked
PassengerId                    ...                          
119                 0       1  ...          B58 B60        C
259                 1       1  ...              NaN        C
300                 1       1  ...          B58 B60        C
312                 1       1  ...  B57 B59 B63 B66        C
378                 0       1  ...              C82        C

[5 rows x 11 columns]

**Можно отсортировать этих людей по убыванию платы за билет.**

In [ ]:
titanic[(titanic["Embarked"] == "C") & (titanic["Fare"] > 200)].sort_values(
    by="Fare", ascending=False
).head()

Survived  Pclass  ...            Cabin Embarked
PassengerId                    ...                          
259                 1       1  ...              NaN        C
680                 1       1  ...      B51 B53 B55        C
738                 1       1  ...             B101        C
312                 1       1  ...  B57 B59 B63 B66        C
743                 1       1  ...  B57 B59 B63 B66        C

[5 rows x 11 columns]

**Пример создания признака.**

In [ ]:
def age_category(age):
    """
    < 30 -> 1
    >= 30, <55 -> 2
    >= 55 -> 3
    """
    if age < 30:
        return 1
    elif age < 55:
        return 2
    else:
        return 3

In [ ]:
age_categories = [age_category(age) for age in titanic.Age]

In [ ]:
titanic["Age_category"] = age_categories

**Другой способ – через apply.**

In [ ]:
titanic["Age_category"] = titanic["Age"].apply(age_category)

#Вопросы для изучения:
**1. Сколько мужчин / женщин находилось на борту?**

In [ ]:
# Ваш код здесь

**2. Выведите распределение переменной `Pclass` (социально-экономический статус) и это же распределение, только для мужчин / женщин по отдельности. Сколько было мужчин 2-го класса?**

* 104
* 108
* 112
* 125

In [ ]:
# Ваш код здесь

**3. Каковы медиана и стандартное отклонение платежей (Fare)? Округлите до 2 десятичных знаков.**

* Медиана – 14.45, стандартное отклонение – 49.69
* Медиана – 15.1, стандартное отклонение – 12.15
* Медиана – 13.15, стандартное отклонение – 35.3
* Медиана – 17.43, стандартное отклонение – 39.1

In [ ]:
# Ваш код здесь

**4. Правда ли, что люди моложе 30 лет выживали чаще, чем люди старше 60 лет? Каковы доли выживших в обеих группах?**
* 22.7% среди молодых и 40.6% среди старых
* 40.6% среди молодых и 22.7% среди старых
* 35.3% среди молодых и 27.4% среди старых
* 27.4% среди молодых и 35.3% среди старых

In [ ]:
# Ваш код здесь

**5. Правда ли, что женщины выживали чаще мужчин? Каковы доли выживших в обеих группах?**

* 30.2% среди мужчин и 46.2% среди женщин
* 35.7% среди мужчин и 74.2% среди женщин
* 31.8% среди мужчин и 68.1% среди женщин
* 18.9% среди мужчин и 74.2% среди женщин

In [ ]:
# Ваш код здесь

**6. Найдите самое популярное имя среди пассажиров Титаника мужского пола?**

* Charles
* Thomas
* William
* John

In [ ]:
# Ваш код здесь

**7. Сравните распределение возраста у спасенных и у погибших. Средний возраст погибших выше, верно?**

* Да
* Нет

In [ ]:
# Ваш код здесь

**8. Как отличается средний возраст мужчин / женщин в зависимости от класса обслуживания? Выберите верные утверждения:**

* В среднем мужчины 1-го класса старше 40 лет
* В среднем женщины 1-го класса старше 40 лет
* Мужчины всех классов в среднем старше женщин того же класса
* В среднем люди в 1 классе старше, чем во 2-ом, а те старше представителей 3-го класса

In [ ]:
# Ваш код здесь